gdb 7.8.2 doesn't build with python 3.7

When the default python resolves to python 3.7 (e.g. in fedora 32), building the toolchain fails due to gdb 7.8.2 not compatible with python 3.7:

```
/home/dciabrin/dam/ngdk/ngdevkit-toolchain/toolchain/gdb-7.8.2/gdb/python/python.c: In function ‘_initialize_python’:
/home/dciabrin/dam/ngdk/ngdevkit-toolchain/toolchain/gdb-7.8.2/gdb/python/python.c:1691:3: error: too few arguments to function ‘_PyImport_FixupBuiltin’
 1691 |   _PyImport_FixupBuiltin (gdb_module, "_gdb");
      |   ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.7m/Python.h:145,
                 from /home/dciabrin/dam/ngdk/ngdevkit-toolchain/toolchain/gdb-7.8.2/gdb/python/python-internal.h:97,
                 from /home/dciabrin/dam/ngdk/ngdevkit-toolchain/toolchain/gdb-7.8.2/gdb/python/python.c:94:
/usr/include/python3.7m/import.h:112:17: note: declared here
  112 | PyAPI_FUNC(int) _PyImport_FixupBuiltin(
      |                 ^~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:2369: python.o] Error 1
make[3]: Leaving directory '/home/dciabrin/dam/ngdk/ngdevkit-toolchain/build/nggdb/gdb'
make[2]: *** [Makefile:8668: all-gdb] Error 2
make[2]: Leaving directory '/home/dciabrin/dam/ngdk/ngdevkit-toolchain/build/nggdb'
make[1]: *** [Makefile:832: all] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@ LOCAL_PACKAGE_DIR=
 SRC_BINUTILS=binutils-2.32
 SRC_GCC=gcc-5.5.0
 SRC_NEWLIB=newlib-1.14.0
-SRC_GDB=gdb-7.8.2
+SRC_GDB=gdb-8.3.1
 SRC_SDCC=sdcc-src-3.7.0
 
 TOOLCHAIN=ngbinutils nggcc ngnewlib ngsdcc nggdb
@@ -53,7 +53,7 @@ download-toolchain: \
 	toolchain/$(SRC_BINUTILS).tar.bz2 \
 	toolchain/$(SRC_GCC).tar.xz \
 	toolchain/$(SRC_NEWLIB).tar.gz \
-	toolchain/$(SRC_GDB).tar.gz \
+	toolchain/$(SRC_GDB).tar.xz \
 	toolchain/$(SRC_SDCC).tar.bz2
 
 unpack-toolchain: \
@@ -80,7 +80,7 @@ toolchain/$(SRC_GCC).tar.xz:
 toolchain/$(SRC_NEWLIB).tar.gz:
 	curl -L $(NEWLIB_MIRROR)/newlib/$(notdir $@) -o $@
 
-toolchain/$(SRC_GDB).tar.gz:
+toolchain/$(SRC_GDB).tar.xz:
 	curl -L $(GNU_MIRROR)/gdb/$(notdir $@) -o $@
 
 toolchain/$(SRC_SDCC).tar.bz2:
@@ -89,7 +89,7 @@ else
 toolchain/$(SRC_BINUTILS).tar.bz2 \
 toolchain/$(SRC_GCC).tar.xz \
 toolchain/$(SRC_NEWLIB).tar.gz \
-toolchain/$(SRC_GDB).tar.gz \
+toolchain/$(SRC_GDB).tar.xz \
 toolchain/$(SRC_SDCC).tar.bz2:
 	cp $(LOCAL_PACKAGE_DIR)/$(notdir $@) $@
 endif
@@ -98,7 +98,7 @@ endif
 toolchain/$(SRC_BINUTILS): toolchain/$(SRC_BINUTILS).tar.bz2
 toolchain/$(SRC_GCC): toolchain/$(SRC_GCC).tar.xz
 toolchain/$(SRC_NEWLIB): toolchain/$(SRC_NEWLIB).tar.gz
-toolchain/$(SRC_GDB): toolchain/$(SRC_GDB).tar.gz
+toolchain/$(SRC_GDB): toolchain/$(SRC_GDB).tar.xz
 toolchain/sdcc: toolchain/$(SRC_SDCC).tar.bz2
 
 

